File: day1.py
def calcFuel(m):
    return (m // 3) -2

def calcFuelRec(m):
    fuel = (m // 3) -2
    if fuel <= 0:
        return 0
    else:
        return fuel + calcFuelRec(fuel)

def day1():
    with open("input1.txt", "r") as f:
        lines = f.readlines()
        masses = list(map(int, lines))
        s = sum(map(calcFuel, masses))
        s2 = sum(map(calcFuelRec, masses))
    print(f"part1: {s}")
    print(f"part2: {s2}")


if __name__ == '__main__':
    day1()

File: day1.hs
module Day1 where
import System.IO

calcFuel :: Integer -> Integer
calcFuel m = (m `div` 3) - 2

calcFuelRec :: Integer -> Integer
calcFuelRec m = if res <= 0 then 0 else res + calcFuelRec res
                where res = (m `div` 3) - 2 
    

f :: [String] -> [Integer]
f = map read

main = do
    handle <- openFile "input1.txt" ReadMode
    contents <- hGetContents handle
    let masses = f (lines contents)
        fuel = sum $ fmap calcFuel masses
        fuelRec = sum $ fmap calcFuelRec masses
    print fuel
    print fuelRec
    hClose handle
    return ()
    
